### PR TITLE
Port to net standard

### PR DESCRIPTION
**Pull request by Mun1z**
Creation of project in .net standard.
Link the files from Telepathy project to TelepathyStandard project.
Upgrade in Nunit version and addition of test adapter in the tests project.

### CODE REVIEW

**Comment by vis2k** [comment]
@paulpach what do you think about this one?
my knowledge of .net / c# is a bit limited

**Comment by paulpach** [comment]
@Mun1z   doesn't this break the assembly for .Net 3.5?

**Comment by Mun1z** [comment]
@paulpach  It does not break, as it is a new standard .NET type project added in the solution, this project uses the same project classes in .net 3.5, they only have advantages, with this new standard .Net project now you can use telepathy in projects type .net core, .net standard, xamarin and any other with .net 4.x. in addition to maintaining compatibility with earlier versions using 3.5.

**Comment by paulpach** [comment]
Call me crazy,  but is it not just a matter of
```
msbuild.exe MyProj.proj /p:TargetFrameworkVersion=netstandard2.0
```

Also,  why is nunit update in this pull request?  what does that have to do with .net update?

**Comment by Mun1z** [comment]
No @paulpach , .net standard is not a new version of the .net framework, it is a different framework, just like .net core, the .net standard it was created for code compatibility between .net applications and .net core applications, when we have developed .net libraries now we have to take into account all the different .net frameworks.

I do not know if you know but a few years ago .net was rewritten to be multiplatform, it is called .net core, the .net core besides being multiplatform it is also about 10 times faster than the old .NET, sometimes the perfomance is the same as that of c ++.

As it is a different framework then .net core does not support old .net nor the reverse, so microsoft created the .net standard to perform this code sharing between different .net frameworks.

A standard .NET library will work on the old .NET, .net core, unity, xamarin, blazor, mono, and any other frameworks that use C #, f #, q #, any microsoft technology. that's god's stuff.

**Comment by paulpach** [comment]
@Mun1z  what about multi target https://oren.codes/2017/01/04/multi-targeting-the-world-a-single-project-to-rule-them-all/
also read this:  https://markheath.net/post/supporting-net-standard-and-net-35

all we would need to do is add this to the .csproj file:
```<TargetFrameworks>netstandard2.0;net35</TargetFrameworks>```

and I still don't understand what nunit update is doing here.

File: Telepathy.Tests/packages.config [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="NUnit" version="3.10.1" targetFramework="net47" />
+  <package id="NUnit" version="3.11.0" targetFramework="net47" />
+  <package id="NUnit3TestAdapter" version="3.10.0" targetFramework="net47" />
```
Comment (paulpach) [inline Telepathy.Tests/packages.config]: should probably be updated to NUnit3TestAdapter version 3.11.0

File: Telepathy.Tests/Telepathy.Tests.csproj [hunk 2]
```diff
@@ -44,4 +47,11 @@
     </ProjectReference>
   </ItemGroup>
   <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
-</Project>
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
```
Comment (paulpach) [inline Telepathy.Tests/Telepathy.Tests.csproj]: Is this necessary?   it already gives an error if nunit is missing

**Comment by Mun1z** [comment]
Okay @paulpach , I'm at work now but once I get home I'm testing the csproj approach, I really did not know it might work, the unit test only worked on my machine when I updated the packages so it went up together.

**Comment by paulpach** [comment]
The multi-target branch has been merged.   It should now support .netstandard2.0  I am going to close this as not necessary anymore.   Feel free to reopen if I missed something

**Pull request closed without merging**